src/test/java/net/minecraftforge/debug/client/AnimateEntityLivingTest.java

```diff
@@ -0,0 +1,190 @@
+            {
+                return entity.getVehicle() != null && entity.getVehicle().getType() == EntityType.PIG;
+            }
+
+            @Override
+            public void apply(Player player, ModelComponent root, Context context)
+            {
+                ModelPart leftLeg = root.get("left_leg");
+                ModelPart rightLeg = root.get("right_leg");
+                leftLeg.xRot = (float) Math.toRadians(-90F);
+                rightLeg.xRot = (float) Math.toRadians(-90F);
+                leftLeg.yRot = (float) Math.toRadians(-45F);
+                rightLeg.yRot = (float) Math.toRadians(45F);
+                leftLeg.zRot = 0;
+                rightLeg.zRot = 0;
+                root.get("left_pants").copyFrom(leftLeg);
+                root.get("right_pants").copyFrom(rightLeg);
+            }
+        });
+
+        // This should overlay an animation on top of the animation defined above
+        event.addAnimation(EntityType.PLAYER, new EntityAnimation<>(EntityAnimation.Mode.PASSIVE, EntityAnimation.Priority.DEFAULT)
+        {
+            @Override
+            public boolean canRun(Player entity)
+            {
+                return entity.getVehicle() != null && entity.getVehicle().getType() == EntityType.PIG && entity.getMainHandItem().getItem() == Items.PIG_SPAWN_EGG;
+            }
+
+            @Override
+            public void apply(Player player, ModelComponent root, Context context)
+            {
+                ModelPart leftLeg = root.get("left_leg");
+                ModelPart rightLeg = root.get("right_leg");
+                leftLeg.yRot = (float) Math.toRadians(-90F);
+                rightLeg.yRot = (float) Math.toRadians(90F);
+                root.get("left_pants").copyFrom(leftLeg);
+                root.get("right_pants").copyFrom(rightLeg);
+            }
+        });
+    }
+
+    private void setupZombieAnimationTest(EntityRenderersEvent.AddAnimations event)
+    {
+        event.addAnimation(EntityType.ZOMBIE, new EntityAnimation<>(EntityAnimation.Mode.PASSIVE, EntityAnimation.Priority.FIRST)
+        {
+            @Override
+            public boolean canRun(Zombie entity)
+            {
+                return entity.isOnFire();
+            }
+
+            @Override
+            public void apply(Zombie entity, ModelComponent root, Context context)
+            {
+                float rotation = (entity.tickCount + context.partialTicks()) * 20F;
+                root.get("right_arm").xRot = (float) Math.toRadians(rotation);
+                root.get("left_arm").xRot = (float) Math.toRadians(rotation + 180F);
+            }
+        });
+    }
+
+    // Creates a test to show active animations cancelling later priority active animations
+    private void setupArmorStandAnimationTest(EntityRenderersEvent.AddAnimations event)
+    {
+        event.addAnimation(EntityType.ARMOR_STAND, new EntityAnimation<>(EntityAnimation.Mode.ACTIVE, EntityAnimation.Priority.FIRST)
```

Comment:
For completeness' sake, please add a comment explaining what these two animations do, as you've done for the others.

Reply:
Fixed. Included more comments and renamed method to clearly indicate test.